File: cole/day35_longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        a = 0
        max_interval = 0
        window = set()
        for i, c in enumerate(s):
            if c in window:
                max_interval = max(max_interval, len(window))
                while s[a] != c:
                    window.remove(s[a])
                    a += 1
                window.remove(s[a])
                a += 1
            window.add(c)
        max_interval = max(max_interval, len(window))
        return max_interval

File: cole/day39_longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.c
#define MAX(a, b) (a > b ? a : b)

int lengthOfLongestSubstring(char * s) {
    int max = 0, x = 0, len = strlen(s);
    char *start = s;
    char h[256] = {'\0'};
    while (*s) {
        if (h[*s] != '\0') {
            max = MAX(max, s - start);
            while (*start != *s) {
                h[*(start++)] = '\0';
            }
            ++start;
        }
        else h[*s] = *s;
        ++s;
    }
    max = MAX(max, s - start);
    return max;
}
